app/shared/ui-settings/src/commonMain/kotlin/ui/settings/tabs/theme/ThemePreferences.kt

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (C) 2024-2025 OpenAni and contributors.
+ *
+ * 此源代码的使用受 GNU AFFERO GENERAL PUBLIC LICENSE version 3 许可证的约束, 可以在以下链接找到该许可证.
+ * Use of this source code is governed by the GNU AGPLv3 license, which can be found at the following link.
+ *
+ * https://github.com/open-ani/ani/blob/main/LICENSE
+ */
+
+package me.him188.ani.app.ui.settings.tabs.theme
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.FlowRow
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.rounded.DarkMode
+import androidx.compose.material.icons.rounded.HdrAuto
+import androidx.compose.material.icons.rounded.LightMode
+import androidx.compose.material3.Icon
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.alpha
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.toArgb
+import com.materialkolor.hct.Hct
+import me.him188.ani.app.data.models.preference.DarkMode
+import me.him188.ani.app.data.models.preference.ThemeSettings
+import me.him188.ani.app.ui.foundation.LocalPlatform
+import me.him188.ani.app.ui.settings.components.ColorButton
+import me.him188.ani.app.ui.settings.framework.SettingsState
+import me.him188.ani.app.ui.settings.framework.components.DropdownItem
+import me.him188.ani.app.ui.settings.framework.components.SettingsScope
+import me.him188.ani.app.ui.settings.framework.components.SwitchItem
+import me.him188.ani.utils.platform.isAndroid
+import me.him188.ani.utils.platform.isDesktop
+
+private val colorList = ((4..10) + (1..3)).map { it * 35.0 }.map { Color(Hct.from(it, 40.0, 40.0).toInt()) }
+
+@Composable
+fun SettingsScope.ThemeGroup(
+    state: SettingsState<ThemeSettings>,
+) {
+    val themeSettings by state
+
+    Group(title = { Text("主题") }) {
+        AnimatedVisibility(
+            LocalPlatform.current.isDesktop() || LocalPlatform.current.isAndroid(),
+        ) {
```

Comment:
```suggestion
        if (LocalPlatform.current.isDesktop() || LocalPlatform.current.isAndroid()) {
```

这里用 if 就行了, 因为 platform 不会动态变更

Reply:
这个可能是导致闪烁的原因

Reply:
经测试无效。已经在 Known Problems 里了，见 pr 第一条。我不打算修，问 gpt 无果。